src/listeners/join-leave-message.ts

```diff
@@ -0,0 +1,51 @@
+import { Client, Colors, EmbedBuilder, GuildMember, PartialGuildMember, userMention } from "discord.js"
+
+const getTargetChannel = async (member: GuildMember | PartialGuildMember) => {
+    const targetChannel = await member.guild.channels.fetch(process.env.GATEWAY_CHANNEL!)
+
+    if (!targetChannel?.isTextBased()) {
+        console.error('Gateway channel is not a text channel!')
+        return undefined
+    }
+
+    return targetChannel
+}
+
+const handleGuildMemberJoin = async (member: GuildMember) => {
+    const targetChannel = await getTargetChannel(member)
+
+    if (!targetChannel) return
+    
+    await targetChannel.send({
+        content: `Welcome ${userMention(member.id)}!`,
+        embeds: [
+            new EmbedBuilder({
+                color: Colors.Green,
+                title: `Welcome to ${member.guild.name}!`,
+                description: 'Enjoy your stay!'
+            })
+        ]
+    })
+}
+
+const handleGuildMemberLeave = async (member: GuildMember | PartialGuildMember) => {
+    const targetChannel = await getTargetChannel(member)
+
+    if (!targetChannel) return
+    
+    await targetChannel.send({
+        embeds: [
+            new EmbedBuilder({
+                color: Colors.Red,
+                title: `Goodbye ${member.user.username}!`
+            })
+        ]
+    })
+}
+
+const listener = (client: Client) => {
```

Comment:
It seems more natural to either export the client from index.ts or another file and then import that directly, or export specific listeners that the client will handle (e.g exporting `handleGuildMemberJoin` as `guildMemberAdd`. It is up to you which to choose, just this approach seems necessarily verbose.

If we really wanted a simple API, globs would also be an option. However, at that point it would be better to implement this in djs-fsrouter.